dipy/denoise/enhancement_kernel.pyx

```diff
@@ -0,0 +1,420 @@
+    cdef object sphere
+    cdef object num_threads
+
+    def __init__(self, D33, D44, t, force_recompute=False,
+                 orientations=None, test_mode=False, num_threads=None):
+        """ Compute a look-up table for the contextual
+        enhancement kernel
+
+        Parameters
+        ----------
+        D33 : float
+            Spatial diffusion
+        D44 : float
+            Angular diffusion
+        t : float
+            Diffusion time
+        force_recompute : boolean
+            Always compute the look-up table even if it is available
+            in cache. Default is False.
+        orientations : integer or Sphere object
+            Specify the number of orientations to be used with
+            electrostatic repulsion, or provide a Sphere object.
+            The default sphere is 'repulsion100'.
+        test_mode : boolean
+            Computes the lookup-table in one direction only
+            
+        References
+        ----------
+        [Meesters2016_ISMRM] S. Meesters, G. Sanguinetti, E. Garyfallidis, 
+                             J. Portegies, R. Duits. (2015) Fast implementations 
+                             of contextual PDE’s for HARDI data processing in 
+                             DIPY. ISMRM 2016 conf. (submitted)
+        [DuitsAndFranken2011] R. Duits and E. Franken (2011) Morphological and
+                              Linear Scale Spaces for Fiber Enhancement in 
+                              DWI-MRI. J Math Imaging Vis, 46(3):326-368.
+        [Portegies2015] J. Portegies, G. Sanguinetti, S. Meesters, and R. Duits.
+                        (2015) New Approximation of a Scale Space Kernel on SE(3) 
+                        and Applications in Neuroimaging. Fifth International
+                        Conference on Scale Space and Variational Methods in
+                        Computer Vision
+        [Portegies2015b] J. Portegies, R. Fick, G. Sanguinetti, S. Meesters, 
+                         G. Girard, and R. Duits. (2015) Improving Fiber 
+                         Alignment in HARDI by Combining Contextual PDE flow with 
+                         Constrained Spherical Deconvolution. PLoS One.
+        """
+
+        # save parameters as class members
+        self.D33 = D33
+        self.D44 = D44
+        self.t = t
+
+        # define a sphere
+        if type(orientations) is Sphere:
+            # use the sphere defined by the user
+            sphere = orientations
+        elif type(orientations) is int:
+            # electrostatic repulsion based on number of orientations
+            n_pts = orientations
+            theta = np.pi * np.random.rand(n_pts)
+            phi = 2 * np.pi * np.random.rand(n_pts)
+            hsph_initial = HemiSphere(theta=theta, phi=phi)
+            sphere, potential = disperse_charges(hsph_initial , 5000)
+        else:
+            # use default
+            sphere = get_sphere('repulsion100')
+        self.orientations = sphere.vertices
```

Comment:
```orientations``` in this case refers to the parameter of ```__init__``` and ```self.orientations``` to the class member. I've renamed the class member to ```orientationsList``` to avoid confusion.